### PR TITLE
James 2773 add glowroot to docker images

### PR DESCRIPTION
**Pull request by remk**

### CODE REVIEW

File: dockerfiles/run/guice/cassandra-ldap/destination/admin.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "web": {
+    "bindAddress": "0.0.0.0"
```
Comment (rouazana) [inline dockerfiles/run/guice/cassandra-ldap/destination/admin.json]: localhost would be sufficient, no?

Comment (blackheaven) [reply dockerfiles/run/guice/cassandra-ldap/destination/admin.json]: You can not access it out of their docker container without that

Comment (rouazana) [reply dockerfiles/run/guice/cassandra-ldap/destination/admin.json]: right, this must be configured when launching the image

**Comment by Arsnael** [comment]
Passed the tests but crashed during the dockering steps

Test this please

**Comment by Arsnael** [comment]
test this please

**Comment by trantienduchn** [comment]
What difference does it make compare to our current grafana metrics?

**Comment by chibenwa** [comment]
Wait my awesome upcoming gatling UDS but it's complementary approaches that don't answer the same questions.

Grafana allows nice graphs with a cool history.

Glowroot will allow you to track down resource (CPU, memory, etc...) utilization per code path (think flame graphs)

More likely, you will detect some slow operations with grafana, and you might want to investigate with glowroot.

**Comment by trantienduchn** [comment]
test this please

**Comment by trantienduchn** [comment]
But there is an imap metric created by glowroot which makes me feel confuse

**Comment by rouazana** [comment]
> But there is an imap metric created by glowroot which makes me feel confuse

It's not really a metric, is a configuration allowing to trace particular requests. For example with this you can easily show which Cassandra requests are related to a specific IMAP operation.

**Comment by trantienduchn** [comment]
this looks mergable

**Comment by rouazana** [comment]
> this looks mergable

we need to make it optional first

**Comment by mbaechler** [comment]
> Glowroot will allow you to track down resource (CPU, memory, etc...) utilization per code path (think flame graphs)

Better than that: you get a flamegraph per call (works with jmap and imap)

**Comment by remk** [comment]
glowroot is now optional and can be activated by launching the docker container with the ```GLOWROOT_ACTIVATED``` environment variable set to ```true```

File: README.adoc [hunk 2]
```diff
@@ -262,9 +262,20 @@ If you want to pass additional options to the underlying java command, you can c
...
 
+==== Instrumentation
+You can use Glowroot to instrumentalize James. The provided guice docker files allow a simple way to do it.
+In order to activate Glowroot you need to run the container with the environment variable _GLOWROOT_ACTIVATED_ set to _true_
+and to expose the glowroot instrumentation ui port.
+
+    --env GLOWROOT_ACTIVATED=true -p "4000:4000"
+
+By default, the Glowroot UI is accessible from every machines in the network as defined in the _destination/admin.json_.
+Which you could configure before building the image, if you want to restrict its accessibility to localhost for example.
```
Comment (rouazana) [inline README.adoc]: Or by mapping the 4000 port to the IP of the desired network interface, for example `-p 127.0.0.1:4000:4000".

**Comment by remk** [comment]
```
org.apache.james.mpt.smtp.AwsS3RabbitMQSmtpStarttlsCommandTest
[a978a8604a249cb377fa00aef4dbaf7c1059251c] [INFO] 
[a978a8604a249cb377fa00aef4dbaf7c1059251c] [INFO] Results:
[a978a8604a249cb377fa00aef4dbaf7c1059251c] [INFO] 
[a978a8604a249cb377fa00aef4dbaf7c1059251c] [ERROR] Errors: 
[a978a8604a249cb377fa00aef4dbaf7c1059251c] [ERROR]   ForwardSmtpTest.forwardingAnEmailShouldWork:79 » ConditionTimeout Assertion co...
```

test this please

**Comment by Arsnael** [comment]
```
[ERROR]   MailRepositoryContract.storingAndRemovingMessagesConcurrentlyShouldLeadToConsistentResult:474 » Execution
```
test this please

**Comment by trantienduchn** [comment]
```
[a978a8604a249cb377fa00aef4dbaf7c1059251c] 2019-06-12T03:20:08.296+0000 I STORAGE  [initandlisten] error creating journal dir /data/db/journal boost::filesystem::create_directory: No space left on device: "/data/db/journal"
[a978a8604a249cb377fa00aef4dbaf7c1059251c] 
[a978a8604a249cb377fa00aef4dbaf7c1059251c] 2019-06-12T03:20:08.296+0000 I STORAGE  [initandlisten] exception in initAndListen std::exception: boost::filesystem::create_directory: No space left on device: "/data/db/journal", terminating
[a978a8604a249cb377fa00aef4dbaf7c1059251c] 
[a978a8604a249cb377fa00aef4dbaf7c1059251c] 2019-06-12T03:20:08.296+0000 I CONTROL  [initandlisten] dbexit:  rc: 100
[a978a8604a249cb377fa00aef4dbaf7c1059251c] 
[a978a8604a249cb377fa00aef4dbaf7c1059251c] [INFO] Running org.apache.james.linshare.LinshareBlobExportMechanismTest
[a978a8604a249cb377fa00aef4dbaf7c1059251c] [ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.009 s <<< FAILURE! - in org.apache.james.linshare.LinshareBlobExportMechanismTest
[a978a8604a249cb377fa00aef4dbaf7c1059251c] [ERROR] org.apache.james.linshare.LinshareBlobExportMechanismTest  Time elapsed: 0.009 s  <<< ERROR!
```
test this please

**Comment by remk** [comment]
test this please

**Comment by remk** [comment]
```

[a978a8604a249cb377fa00aef4dbaf7c1059251c] [INFO] 
[a978a8604a249cb377fa00aef4dbaf7c1059251c] [ERROR] Errors: 
[a978a8604a249cb377fa00aef4dbaf7c1059251c] [ERROR]   ForwardSmtpTest.forwardingAnEmailShouldWork:79 » ConditionTimeout Assertion co...
```
test this please

**Comment by Arsnael** [comment]
```
[a978a8604a249cb377fa00aef4dbaf7c1059251c] [ERROR]   RabbitMQHealthCheckTest.<clinit>:33 » NoClassDefFound Could not initialize cla...
[a978a8604a249cb377fa00aef4dbaf7c1059251c] [ERROR]   RabbitMQTest.<clinit>:58 » ExceptionInInitializer
```
test this please

**Comment by remk** [comment]
rebased on master

**Comment by remk** [comment]
test this please

**Comment by remk** [comment]
```
[542306fee95d4cd3d88758922cb7846e35b2c1ea] [ERROR] Failed to execute goal on project apache-james-mailbox-lucene: Could not resolve dependencies for project org.apache.james:apache-james-mailbox-lucene:bundle:3.4.0-SNAPSHOT: Failed to collect dependencies at org.apache.lucene:lucene-analyzers:jar:3.6.2: Failed to read artifact descriptor for org.apache.lucene:lucene-analyzers:jar:3.6.2: Could not transfer artifact org.apache.lucene:lucene-analyzers:pom:3.6.2 from/to central (https://repo.maven.apache.org/maven2): Failed to transfer file: https://repo.maven.apache.org/maven2/org/apache/lucene/lucene-analyzers/3.6.2/lucene-analyzers-3.6.2.pom. Return code is: 502 , ReasonPhrase:Gateway Error. -> [Help 1]
```
test this please

**Comment by chibenwa** [comment]
```
SmtpRandomStoringTest.sendingOneHundredMessagesShouldBeRandomlyAssignedToEveryMailboxesOfEveryUsers:156 » ConditionTimeout
```

test this please

**Comment by remk** [comment]
No more logs access.
test this please

**Comment by rouazana** [comment]
```
[542306fee95d4cd3d88758922cb7846e35b2c1ea] Step 10/19 : ADD destination/glowroot/glowroot.jar /root/glowroot/glowroot.jar
[542306fee95d4cd3d88758922cb7846e35b2c1ea] ADD failed: stat /var/lib/docker/tmp/docker-builder283601332/destination/glowroot/glowroot.jar: no such file or directory
```
seems related :)

**Comment by blackheaven** [comment]
```
 [INFO] Apache James :: Mailbox :: Event :: RabbitMQ implementation FAILURE [04:21 min]
```

test this please

**Comment by rouazana** [comment]
CI updated, test this please

**Comment by mbaechler** [comment]
it looks like we had a github failure.

test this please

**Comment by chibenwa** [comment]
```
[af422cb80f026ddebfd9a20d9549a128b9a21430] Step 11/20 : COPY destination/glowroot/lib/glowroot-embedded-collector.jar /root/glowroot/lib/glowroot-embedded-collector.jar
[af422cb80f026ddebfd9a20d9549a128b9a21430] COPY failed: Forbidden path outside the build context: destination/glowroot/lib/glowroot-embedded-collector.jar ()
```

**Comment by remk** [comment]
```
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] 2019-06-19 09:08:29,839 WARN  [elastic-12] - [org.apache.james.mailetcontainer.impl.camel.CamelProcessor]- Encountered error while executing mailet org.apache.james.transport.mailets.SpamAssassin@50d94369. Ignoring it.
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] javax.mail.MessagingException: Error communicating with spamd on localhost:783
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] 	at org.apache.james.spamassassin.SpamAssassinInvoker.scanMailWithAdditionalHeaders(SpamAssassinInvoker.java:145)
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] 	at org.apache.james.spamassassin.SpamAssassinInvoker.lambda$scanMail$0(SpamAssassinInvoker.java:100)
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] 	at com.github.fge.lambdas.supplier.SupplierChainer.lambda$sneakyThrow$195(SupplierChainer.java:74)
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] 	at org.apache.james.metrics.api.MetricFactory.runPublishingTimerMetric(MetricFactory.java:35)
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] 	at org.apache.james.spamassassin.SpamAssassinInvoker.scanMail(SpamAssassinInvoker.java:97)
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] 	at org.apache.james.transport.mailets.SpamAssassin.querySpamAssassin(SpamAssassin.java:111)
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] 	at org.apache.james.transport.mailets.SpamAssassin.lambda$service$1(SpamAssassin.java:106)
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] 	at com.github.fge.lambdas.consumers.ConsumerChainer.lambda$sneakyThrow$9(ConsumerChainer.java:73)
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] 	at com.google.common.collect.ImmutableList.forEach(ImmutableList.java:407)
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] 	at org.apache.james.transport.mailets.SpamAssassin.service(SpamAssassin.java:105)
```

test this please

**Comment by remk** [comment]
```
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] [ERROR] Tests run: 102, Failures: 0, Errors: 1, Skipped: 1, Time elapsed: 232.252 s <<< FAILURE! - in org.apache.james.mailbox.events.RabbitMQEventBusTest
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] [ERROR] dispatchShouldWorkAfterNetworkIssuesForOldRegistration  Time elapsed: 55.252 s  <<< ERROR!
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] org.awaitility.core.ConditionTimeoutException: 
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] Assertion condition defined as a lambda expression in org.apache.james.mailbox.events.RabbitMQEventBusTest that uses org.apache.james.mailbox.events.MailboxListener 
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] Wanted but not invoked:
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] mailboxListener.event(
[7465d7b5cd81e4c778c844bad669bd6f336a7fca]     org.apache.james.mailbox.events.MailboxListener$MailboxAdded@bf1a1130
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] );
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] -> at org.apache.james.mailbox.events.RabbitMQEventBusTest.lambda$assertThatListenerReceiveOneEvent$2(RabbitMQEventBusTest.java:685)
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] Actually, there were zero interactions with this mock.
[7465d7b5cd81e4c778c844bad669bd6f336a7fca]  within 30 seconds.
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] 	at org.apache.james.mailbox.events.RabbitMQEventBusTest$LifeCycleTest$SingleEventBus$DispatchingWhenNetWorkIssue.dispatchShouldWorkAfterNetworkIssuesForOldRegistration(RabbitMQEventBusTest.java:366)
```

test this please

**Comment by remk** [comment]
```
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] [ERROR] Errors: 
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] [ERROR]   LinshareBlobExportMechanismProvidingTest.<clinit>:29 » ExceptionInInitializer
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] [INFO] 
```


```
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] 12:52:28.678 [ERROR] o.a.j.j.JMAPConfigurationStartUpCheck - Cannot load security key from jmap configuration
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] java.security.UnrecoverableKeyException: Password verification failed
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] 	at sun.security.provider.JavaKeyStore.engineLoad(JavaKeyStore.java:778)
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] 	... 19 common frames omitted
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] Wrapped by: java.io.IOException: Keystore was tampered with, or password was incorrect
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] 	at sun.security.provider.JavaKeyStore.engineLoad(JavaKeyStore.java:780)
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] 	at sun.security.provider.JavaKeyStore$JKS.engineLoad(JavaKeyStore.java:56)
```

really ?

test this please

**Comment by remk** [comment]
```
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] Get:64 http://security.debian.org/debian-security/ jessie/updates/main libisccfg-export90 amd64 1:9.9.5.dfsg-9+deb8u17 [41.7 kB]
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] Get:65 http://security.debian.org/debian-security/ jessie/updates/main libirs-export91 amd64 1:9.9.5.dfsg-9+deb8u17 [39.4 kB]
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] Get:66 http://security.debian.org/debian-security/ jessie/updates/main libx11-data all 2:1.6.2-3+deb8u2 [126 kB]
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] Get:67 http://security.debian.org/debian-security/ jessie/updates/main libx11-6 amd64 2:1.6.2-3+deb8u2 [729 kB]
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] Get:68 http://security.debian.org/debian-security/ jessie/updates/main krb5-locales all 1.12.1+dfsg-19+deb8u5 [2649 kB]
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] Get:69 http://security.debian.org/debian-security/ jessie/updates/main openssh-client amd64 1:6.7p1-5+deb8u8 [696 kB]
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] Get:70 http://security.debian.org/debian-security/ jessie/updates/main openssl amd64 1.0.1t-1+deb8u11 [665 kB]
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] Get:71 http://security.debian.org/debian-security/ jessie/updates/main ca-certificates all 20141019+deb8u4 [185 kB]
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] Get:72 http://security.debian.org/debian-security/ jessie/updates/main git-man all 1:2.1.4-2.1+deb8u7 [1269 kB]
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] Get:73 http://security.debian.org/debian-security/ jessie/updates/main git amd64 1:2.1.4-2.1+deb8u7 [3709 kB]
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] Get:74 http://security.debian.org/debian-security/ jessie/updates/main rsync amd64 3.1.1-3+deb8u2 [390 kB]
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] [91mE: Failed to fetch http://security.debian.org/debian-security/pool/updates/main/b/bind9/libisc-export95_9.9.5.dfsg-9+deb8u17_amd64.deb  Connection failed [IP: 212.211.132.250 80]
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] 
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] [0mFetched 21.6 MB in 2min 5s (172 kB/s)
```
it's getting old.

test this please

**Comment by remk** [comment]
```
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] Get:73 http://security.debian.org/debian-security/ jessie/updates/main git amd64 1:2.1.4-2.1+deb8u7 [3709 kB]
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] Get:74 http://security.debian.org/debian-security/ jessie/updates/main rsync amd64 3.1.1-3+deb8u2 [390 kB]
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] [91mE: Failed to fetch http://security.debian.org/debian-security/pool/updates/main/b/bind9/libisc-export95_9.9.5.dfsg-9+deb8u17_amd64.deb  Connection failed [IP: 212.211.132.250 80]
```
test this please

**Comment by Arsnael** [comment]
```
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] [ERROR] Error occurred in starting fork, check output in log
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] [ERROR] Process Exit Code: 100
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] [ERROR] Crashed tests:
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] [ERROR] org.apache.james.webadmin.routes.EventDeadLettersRoutesTest$RedeliverSingleEvent
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] [ERROR] org.apache.maven.surefire.booter.SurefireBooterForkException: The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
```
test this please

**Comment by chibenwa** [comment]
```
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] 2019-06-19 15:23:19,815 ERROR [Thread-11] - [spark.Spark]- ignite failed
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] java.net.BindException: Address already in use
```

test this please

**Comment by Arsnael** [comment]
```
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] [ERROR] Errors: 
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] [ERROR]   LinshareBlobExportMechanismProvidingTest.<clinit>:29 » ExceptionInInitializer
```
I start to think that the CI god hates you @remk :)

test this please

**Comment by remk** [comment]
```
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] Step 8/20 : ADD destination/conf /root/conf
[7465d7b5cd81e4c778c844bad669bd6f336a7fca]  ---> 66ba2439d949
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] Step 9/20 : ADD destination/glowroot/plugins /root/glowroot/plugins
[7465d7b5cd81e4c778c844bad669bd6f336a7fca]  ---> 1b268bc64906
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] Step 10/20 : ADD destination/glowroot/glowroot.jar /root/glowroot/glowroot.jar
[7465d7b5cd81e4c778c844bad669bd6f336a7fca]  ---> 6fcad0f826dc
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] Step 11/20 : ADD destination/glowroot/lib/glowroot-embedded-collector.jar /root/glowroot/lib/glowroot-embedded-collector.jar
[7465d7b5cd81e4c778c844bad669bd6f336a7fca] ADD failed: Forbidden path outside the build context: destination/glowroot/lib/glowroot-embedded-collector.jar ()
```
it's a real and strange error on the ci.
it works on my local machine, i don't know why it fail on the ci.
The build context is good and the same as the previous files...
Maybe a difference in docker version ? (Docker version 18.09.6, build 481bc77 on my local machine)

**Comment by remk** [comment]
In the hope to run on ci 02, 03 or 04 which we just updated to the latest docker version
test this please

**Comment by trantienduchn** [comment]
if it still fails, can you just try to add the `glowroot-embedded-collector.jar` to outside of the `lib` directory to see what happen? just trying, i have no idea about that failure

**Comment by remk** [comment]
With the logs added, it's because the files destination/glowroot/plugins, destination/glowroot/admin.json and destination/run_james.sh couldn't be found.

I added some logs in a pr in the jenkins project to investigate this : https://github.com/linagora/james-jenkins/pull/165

**Comment by remk** [comment]
test this please

**Comment by remk** [comment]
test this please

**Comment by remk** [comment]
test this please

**Comment by remk** [comment]
test this please

**Comment by trantienduchn** [comment]
test this please

**Comment by rouazana** [comment]
test this please (CI updated)

**Comment by remk** [comment]
```
[b65f27740045f6752a6ccc098a62c112ed5a0c68] Running: Shell Script
[b65f27740045f6752a6ccc098a62c112ed5a0c68] [807] Running shell script
[b65f27740045f6752a6ccc098a62c112ed5a0c68] + docker run --detach=true --name=james-server-cassandra-rabbitmq-james-2773-201475e8-db00-4b49-b4ec-955dc57fdffd --hostname james.linagora.com --expose=143 --expose=587 --publish-all=true --link cassandra-james-2773-201475e8-db00-4b49-b4ec-955dc57fdffd:cassandra --link elasticSearch-james-2773-201475e8-db00-4b49-b4ec-955dc57fdffd:elasticsearch --link rabbitmq-james-2773-201475e8-db00-4b49-b4ec-955dc57fdffd:rabbitmq --link swift-james-2773-201475e8-db00-4b49-b4ec-955dc57fdffd:swift james-server-cassandra-rabbitmq-james-2773-201475e8-db00-4b49-b4ec-955dc57fdffd
[b65f27740045f6752a6ccc098a62c112ed5a0c68] c8e89161ee0cd6cbe3744dc67b19673deb2250541ad3fb9591d8adf43d55f402
[b65f27740045f6752a6ccc098a62c112ed5a0c68] docker: Error response from daemon: Cannot link to a non running container: /rabbitmq-james-2773-201475e8-db00-4b49-b4ec-955dc57fdffd AS /james-server-cassandra-rabbitmq-james-2773-201475e8-db00-4b49-b4ec-955dc57fdffd/rabbitmq.
[b65f27740045f6752a6ccc098a62c112ed5a0c68] Running: Shell Script
[b65f27740045f6752a6ccc098a62c112ed5a0c68] [807] Running shell script
```

test this please

**Comment by remk** [comment]
```
[b65f27740045f6752a6ccc098a62c112ed5a0c68] cbf06c9710e8f0da7a469fdc92f593b5e6dbbbcabbe8ad55612e7ff541d78f7f
[b65f27740045f6752a6ccc098a62c112ed5a0c68] docker: Error response from daemon: Cannot link to a non running container: /cassandra-james-2773-8f9427d4-e1cf-4d48-8513-297c341be513 AS /james-server-cassandra-rabbitmq-james-2773-8f9427d4-e1cf-4d48-8513-297c341be513/cassandra.
```

test this please

**Comment by chibenwa** [comment]
```
[b65f27740045f6752a6ccc098a62c112ed5a0c68] docker: Error response from daemon: Cannot link to a non running container: /rabbitmq-james-2773-f57b653d-e965-49a4-92fd-286745e44b05 AS /james-server-cassandra-rabbitmq-james-2773-f57b653d-e965-49a4-92fd-286745e44b05/rabbitmq.
```

**Comment by remk** [comment]
```
[b65f27740045f6752a6ccc098a62c112ed5a0c68] [816] Running shell script
[b65f27740045f6752a6ccc098a62c112ed5a0c68] + docker logs cassandra-james-2773-f57b653d-e965-49a4-92fd-286745e44b05
[b65f27740045f6752a6ccc098a62c112ed5a0c68] sed: couldn't write 75 items to stdout: No space left on device
[b65f27740045f6752a6ccc098a62c112ed5a0c68] Running: Shell Script
```

test this please

**Comment by remk** [comment]
```
[b65f27740045f6752a6ccc098a62c112ed5a0c68] + docker exec james-server-jpa-james-2773-094a4723-eaf3-418e-bafc-e692bc9d88b1 java -jar /root/james-cli.jar -h 127.0.0.1 listusers
[b65f27740045f6752a6ccc098a62c112ed5a0c68] Error response from daemon: Container 86f25ae896bd6fc045352fb0a565fccb76681e4d944d4a5b8a421556abf85d82 is not running
[b65f27740045f6752a6ccc098a62c112ed5a0c68] Running: Shell Script
[b65f27740045f6752a6ccc098a62c112ed5a0c68] [836] Running shell script
[b65f27740045f6752a6ccc098a62c112ed5a0c68] + docker logs james-server-jpa-james-2773-094a4723-eaf3-418e-bafc-e692bc9d88b1
[b65f27740045f6752a6ccc098a62c112ed5a0c68] Error occurred during initialization of VM
[b65f27740045f6752a6ccc098a62c112ed5a0c68] agent library failed to init: instrument
[b65f27740045f6752a6ccc098a62c112ed5a0c68] Error opening zip file or JAR manifest missing : /root/glowroot.jar
```

**Comment by remk** [comment]
```
Running: End of Workflow
java.io.IOException: Server returned HTTP response code: 500 for URL: https://api.github.com/repos/linagora/james-project/statuses/7cc2b9902c1f5748fb31fd83a19411ea22c3b32d
```

test this please

**Comment by remk** [comment]
```
java.io.IOException: Server returned HTTP response code: 500 for URL: https://api.github.com/repos/linagora/james-project/statuses/7cc2b9902c1f5748fb31fd83a19411ea22c3b32d
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
```

test this please

**Comment by remk** [comment]
sprint : 

[ddad78b4818ae19a108b20432fab9ba6b895eb90] Step 7/21 : ADD destination/glowroot/glowroot.jar /root/glowroot/glowroot.jar
[ddad78b4818ae19a108b20432fab9ba6b895eb90] ADD failed: stat /var/lib/docker/tmp/docker-builder219605996/destination/glowroot/glowroot.jar: no such file or directory

**Comment by remk** [comment]
test this please

**Comment by remk** [comment]
test this please

**Comment by remk** [comment]
need https://github.com/linagora/james-jenkins/pull/170 to be merged

**Comment by rouazana** [comment]
test this please

**Comment by remk** [comment]
test this please

**Comment by remk** [comment]
test this please

**Comment by remk** [comment]
test this please

**Comment by remk** [comment]
:green_apple:  !!!
I will rebase and remove commit to remove tests

**Pull request merged**